crates/turborepo-lib/src/shim.rs

```diff
@@ -176,50 +178,234 @@ struct PackageJson {
     version: String,
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct LocalTurboState {
-    bin_path: PathBuf,
-    version: String,
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct YarnRc {
+    pnp_unplugged_folder: PathBuf,
 }
 
-impl LocalTurboState {
-    pub fn infer(repo_root: &Path) -> Option<Self> {
-        let local_turbo_path = repo_root.join("node_modules").join(".bin").join({
+impl Default for YarnRc {
+    fn default() -> Self {
+        Self {
+            pnp_unplugged_folder: [".yarn", "unplugged"].iter().collect(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TurboState {
+    bin_path: Option<PathBuf>,
+    version: &'static str,
+    repo_state: Option<RepoState>,
+}
+
+impl Default for TurboState {
+    fn default() -> Self {
+        Self {
+            bin_path: env::current_exe().ok(),
+            version: get_version(),
+            repo_state: None,
+        }
+    }
+}
+
+impl TurboState {
+    pub fn platform_package_name() -> &'static str {
+        const ARCH: &str = {
+            #[cfg(target_arch = "x86_64")]
+            {
+                "64"
+            }
+            #[cfg(target_arch = "aarch64")]
+            {
+                "arm64"
+            }
+            #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+            {
+                "unknown"
+            }
+        };
+
+        const OS: &str = {
+            #[cfg(target_os = "macos")]
+            {
+                "darwin"
+            }
+            #[cfg(target_os = "windows")]
+            {
+                "windows"
+            }
+            #[cfg(target_os = "linux")]
+            {
+                "linux"
+            }
+            #[cfg(not(any(target_os = "macos", target_os = "windows", target_os = "linux")))]
+            {
+                "unknown"
+            }
+        };
+
+        formatcp!("turbo-{}-{}", OS, ARCH)
+    }
+
+    pub fn binary_name() -> &'static str {
```

Comment:
I think this could also be a constant?

Reply:
Optimizations at compile time should inline this just the same so far as I can tell.